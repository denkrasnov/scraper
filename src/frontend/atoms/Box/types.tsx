import { ReactNode } from "react";

import { Color } from "../Colors/types";

type Display =
  | "block"
  | "inline-block"
  | "inline"
  | "flex"
  | "inline-flex"
  | "none";

type FlexDirection = "column" | "row";

type JustifyContent =
  | "flex-start"
  | "flex-end"
  | "center"
  | "space-between"
  | "space-around"
  | "space-evenly";

export type AlignItems =
  | "flex-start"
  | "flex-end"
  | "center"
  | "baseline"
  | "stretch";

type FlexWrap = "wrap" | "nowrap";

type Cursor = "pointer";

export interface BoxProps {
  alignItems?: AlignItems;
  backgroundColor?: Color;
  children?: ReactNode;
  display?: Display;
  flexDirection?: FlexDirection;
  flexWrap?: FlexWrap;
  grow?: number;
  flex?: number;
  justifyContent?: JustifyContent;
  padding?: string;
  margin?: string;
  marginBottom?: string;
  width?: string;
  height?: string;
  minHeight?: string;
  maxWidth?: string;
  cursor?: Cursor;
  marginRight?: string;
  onClick?: () => void;
}
