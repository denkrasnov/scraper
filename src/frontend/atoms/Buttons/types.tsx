import { ReactNode } from "react";

type ButtonType = "submit" | "button" | "reset";

export interface ButtonProps {
  disabled?: boolean;
  children?: ReactNode;
  onClick?: () => void;
  type?: ButtonType;
  transparent?: boolean;
  color?: string;
}
