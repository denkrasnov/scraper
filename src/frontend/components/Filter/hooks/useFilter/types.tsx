export type Filter = { key: string; value: string };

export type UseFilterReturn<T> = [
  T[],
  (checked: boolean, value: string, filterByKey: string) => void
];
