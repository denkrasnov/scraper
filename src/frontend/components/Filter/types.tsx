// TODO: create shared types with /backend
import { ProductName, Article, Channel } from "../../../backend/scrapers/types";
import { Locale } from "../../../types";

export interface FilterProps {
  productName: ProductName;
  options: Channel[];
  items: Article[];
  locale?: Locale;
  onClickLocale?: () => void;
}
