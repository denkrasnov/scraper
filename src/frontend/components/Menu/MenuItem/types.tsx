import { DataAttr } from "~app/common/types";

export interface MenuItemProps {
  src: string;
  text: string;
  onClick: () => void;
  dataAttr?: DataAttr;
}
