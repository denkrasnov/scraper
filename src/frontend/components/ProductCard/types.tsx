import { Article } from "../../../backend/scrapers/types";

export interface ProductCardProps extends Article {}
