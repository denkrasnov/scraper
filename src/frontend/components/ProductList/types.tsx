import { Article } from "../../../backend/scrapers/types";

export interface ProductListProps {
  products: Article[];
}
