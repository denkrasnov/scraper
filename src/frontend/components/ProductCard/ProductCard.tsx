import React, { FC } from "react";

import Image from "~app/atoms/Image";
import Card from "~app/atoms/Card";
import Box from "~app/atoms/Box";
import { H3 } from "~app/atoms/Headings";
import Text from "~app/atoms/Text";
import Divider from "~app/atoms/Divider";
import { ProductCardProps } from "./types";
import { image } from "~app/assets";

const ProductCard: FC<ProductCardProps> = (props) => {
  const { title, imageUrl, price, noImage } = props;

  return (
    <Card>
      <Box flexDirection="column" height="100%">
        <Box height="200px">
          <Image
            alt="product photo"
            src={noImage ? image.tvPlaceholder : imageUrl}
          />
        </Box>
        <Box flexDirection="column" height="100%" padding="0 10px 10px 10px">
          <H3>{title}</H3>
          <Box flex={1} />
          <Divider />
          <Box justifyContent="flex-end" margin="15px 0px 5px 0">
            <Text color="MAIN_RED" fontSize="fs400" fontWeight="fw700">
              lei {price}
            </Text>
          </Box>
        </Box>
      </Box>
    </Card>
  );
};

export default ProductCard;
