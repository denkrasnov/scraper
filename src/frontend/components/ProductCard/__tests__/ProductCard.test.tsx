import { shallow } from "enzyme";

import { Channel } from "../../../../backend/scrapers/types";
import ProductCard from "..";

describe("ProductCard", () => {
  it("should render", () => {
    const component = shallow(
      <ProductCard
        channel={Channel.TV8}
        date="9:00"
        header="__TITLE__"
        imageURL="__IMAGE_URL__"
        newsURL="__URL__"
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("should render when no image url", () => {
    const component = shallow(
      <ProductCard
        channel={Channel.TV8}
        date="9:00"
        header="__TITLE__"
        imageURL=""
        newsURL="__URL__"
      />
    );
    expect(component).toMatchSnapshot();
  });
});
