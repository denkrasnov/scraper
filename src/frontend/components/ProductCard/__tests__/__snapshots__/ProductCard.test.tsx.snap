// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render 1`] = `
<Touchable
  dataAttr={
    Object {
      "data-e2e-id": "productCard",
    }
  }
  href="__URL__"
  productHover={true}
  width="100%"
>
  <Card
    fullHeight={true}
  >
    <Box
      padding="s12"
      width="100%"
    >
      <Box
        height="120px"
        marginRight="s12"
        minHeight="120px"
      >
        <Image
          alt="product photo"
          src="__IMAGE_URL__"
          width="100%"
        />
      </Box>
      <Box
        flexDirection="column"
        width="100%"
      >
        <Box
          alignItems="center"
          flexGrow="gr1"
        >
          <Text>
            <H3>
              __TITLE__
            </H3>
          </Text>
        </Box>
        <Divider />
        <Box
          justifyContent="spaceBetween"
          marginTop="s8"
        >
          <Image
            alt="channel"
            src="https://tv8.md/wp-content/uploads/2020/07/logo-tv8-new-2.png"
            width="30px"
          />
          <Text
            color="BORDER_DARK"
            fontSize="fs16"
            fontWeight="fw700"
          >
            9:00
          </Text>
        </Box>
      </Box>
    </Box>
  </Card>
</Touchable>
`;

exports[`ProductCard should render when no image url 1`] = `
<Touchable
  dataAttr={
    Object {
      "data-e2e-id": "productCard",
    }
  }
  href="__URL__"
  productHover={true}
  width="100%"
>
  <Card
    fullHeight={true}
  >
    <Box
      padding="s12"
      width="100%"
    >
      <Box
        height="120px"
        marginRight="s12"
        minHeight="120px"
      >
        <Box
          alignSelf="center"
        >
          <Text
            color="BORDER_GRAY"
            fontSize="fs24"
            fontWeight="fw700"
          >
            No photo
          </Text>
        </Box>
      </Box>
      <Box
        flexDirection="column"
        width="100%"
      >
        <Box
          alignItems="center"
          flexGrow="gr1"
        >
          <Text>
            <H3>
              __TITLE__
            </H3>
          </Text>
        </Box>
        <Divider />
        <Box
          justifyContent="spaceBetween"
          marginTop="s8"
        >
          <Image
            alt="channel"
            src="https://tv8.md/wp-content/uploads/2020/07/logo-tv8-new-2.png"
            width="30px"
          />
          <Text
            color="BORDER_DARK"
            fontSize="fs16"
            fontWeight="fw700"
          >
            9:00
          </Text>
        </Box>
      </Box>
    </Box>
  </Card>
</Touchable>
`;
